lib/pbench/server/database/models/tracker.py

```diff
@@ -671,10 +671,26 @@ class Metadata(Database.Base):
 
     __tablename__ = "dataset_metadata"
 
-    # Currently defined metadata keys
+    # REINDEX boolean flag to indicate when a dataset should be re-indexed
+    #
+    # {"REINDEX": True}
     REINDEX = "REINDEX"
+
+    # ARCHIVED boolean flag to indicate when a tarball has been archived
+    #
+    # {"ARCHIVED": True}
     ARCHIVED = "ARCHIVED"
+
+    # TARBALL_PATH access path of the dataset tarball. (E.g., we could use this
+    # to record an S3 object store key.) NOT YET USED.
+    #
+    # {"TARBALL_PATH": "/srv/pbench/archive/fs-version-001/ctrl/example__2021.05.21T07.15.27.tar.xz"}
     TARBALL_PATH = "TARBALL_PATH"
+
+    # INDEX_MAP a two-level dict recording the MD5 document IDs for each
+    # Elasticsearch index containing documents for this dataset.
+    #
+    # {"drb.v6.run-data.2021-07": ["MD5"], "drb.v6.run-toc.2021-07": ["MD5-1", "MD5-2"]}
```

Comment:
OK, maybe "two-level dict" was bad phrasing, since I suppose it does imply that the "second level" is also a `dict`. The value of the `INDEX_MAP` key is a serialized JSON document where, as you observe, each key represents a fully-formed Elasticsearch index name with the value being a list of all Elasticsearch `_id` values (document MD5 sums in our usage) for that index.

I tried to create an example that would fit on a single line, and perhaps the use of actual sample Pbench index names was inappropriate in that context; "index1" and "index2" would have been more compact and possibly less confusing in this context. Is this worth changing?

Aside from that, this is intended to loosely represent the *format* but not to precisely represent the *content*; however, yes each dataset will have a single document in the "run" index and one or more entries in the TOC index as well as 0 or **many** in the result document indices and tool-specific indices. The point is that the map lists all indices in which the dataset has documents, and includes the document id of each of those documents.

Reply:
If you end up doing another round of changes for review, you can catch it then.  (Otherwise, in the future, you'll probably know what you meant, so it's all good.  😉)

(I'm going to resolve this conversation so that it doesn't gate merging.)